src/Kanvas/Filesystem/Actions/CreateFilesystemMapperAction.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kanvas\Filesystem\Actions;
+
+use Exception;
+use Kanvas\Filesystem\DataTransferObject\FilesystemMapper;
+use Kanvas\Filesystem\Models\FilesystemMapper as ModelsFilesystemMapper;
+
+class CreateFilesystemMapperAction
+{
+    public function __construct(
+        protected FilesystemMapper $filesystemMapping
+    ) {
+    }
+
+    public function execute(): ModelsFilesystemMapper
+    {
+        $mapperKeys = array_keys($this->filesystemMapping->mapping);
+        if (array_diff($mapperKeys, $this->filesystemMapping->systemModule->browse_fields)) {
```

Comment:
@FredPeal dont know if it has to be the same , maybe you can just have it validate that the field in the browse field exist in what coming in if it doesn't , you fail